File: LunamaroApi/Lunamaroapi/Models/OrderItem.cs
using System.ComponentModel.DataAnnotations.Schema;

namespace Lunamaroapi.Models
{
 public class OrderItem
    {
        public int OrderItemId { get; set; }

        public int UserOrderHeaderId { get; set; }
        [ForeignKey("UserOrderHeaderId")]
        public UserOrderHeader UserOrderHeader { get; set; }

        public int ItemId { get; set; }
        public Item Item { get; set; }

        public int Quantity { get; set; }

        [Column(TypeName = "decimal(18,2)")]
        public decimal UnitPrice { get; set; }

        public decimal TotalPrice => UnitPrice * Quantity;

    }

}

File: LounamaroUI/src/app/Models/order-item.ts
import { Item } from "./item";
import { UserOrderHeader } from "./user-order-header";

export interface OrderItem {
  orderItemId: number;
  userOrderHeaderId: number;
  userOrderHeader?: UserOrderHeader;
  itemId: number;
  item?: Item;
  quantity: number;
  unitPrice: number;
  totalPrice: number;
}
